packages/superset-ui-chart/src/components/reactify.tsx

```diff
@@ -17,6 +17,11 @@ export type ReactifyProps = {
   className?: string;
 };
 
+// TODO: add more React lifecycle callbacks as needed
+export type LifeCycleCallbacks = {
+  willUnmount?: () => void;
```

Comment:
small point -- but I think that these should mirror the actual lifecycle names because
1) devs are already familiar with them
2) some of them start with `componentXXX` and some don't (e.g., `shouldComponentUpdate`). The one's that don't are a little more cryptic when truncated (e.g. `componentUpdate`?), so if/when we add more we'll then have a mix of full names and short names.

Reply:
Good point! Let me rename this to `componentWillUnmount` and I'll adjust the usage in `nvd3` charts in `superset-ui-plugins`.